### PR TITLE
Add a static file service.

### PR DESCRIPTION
**Pull request by sellout**
This serves up URIs under /slamdata relative to the location of the jar. It
also redirects / to /slamdata (assuming no other service matches
the path).

This also adds command-line options to the server (primarily to be able
to choose whether or not starting the server opens a client in the
browser). This meant changing the config file from being the first
argument to being passed with `--config`. I also add a `--port` option
just for convenience.

### CODE REVIEW

File: web/src/main/scala/slamdata/engine/api/fs.scala [hunk 1]
```diff
@@ -284,4 +284,32 @@ class FileSystemApi(fs: FSTable[Backend]) {
...
+
+  val basePath: Task[String] =
+    Task.delay((new File(Server.getClass.getProtectionDomain().getCodeSource().getLocation().toURI().getPath()))
+      .getParentFile()
+      .getPath())
+
+  def fileMediaType(file: String): Option[MediaType] =
+    MediaType.forExtension(file.split('.').last)
+
+  def appService = corsService {
```
Comment (mossprescott) [inline web/src/main/scala/slamdata/engine/api/fs.scala]: I'd rather see a more generic service here, taking filePath as a parameter at the very least. As is, it's making assumptions about how we deploy the front-end.

Comment (sellout) [inline web/src/main/scala/slamdata/engine/api/fs.scala]: Yeah, that’s a good idea.

**Comment by mossprescott** [comment]
@sellout: I'd like to get this merged so I can work on getting both *-ends* running in Jenkins. Did you forget about it?

**Comment by sellout** [comment]
Sorry – was running `ready` when I left the office. Just got back to it now. Should be good once this build passes.

**Comment by mossprescott** [comment]
It's failing on a PUT to the data service that should give a 400. Hmm.

**Comment by sellout** [comment]
That error is very odd, first because I don’t think anything I did affected the data service, but also because I had restarted the build and the first time through it failed with the NPE you’re seeing in #709.

**Pull request merged**